### ISSUE TITLE
4.0.0 Github auth is broken

### ISSUE THREAD

**Issue by mxplusb** [concourse/concourse#2530]
# Bug Report

I'm trying to figure out what is wrong with my GitHub auth on 4.0.0. I can browse to the ATC GUI, click login, get routed to GitHub, but the callback isn't working as intended. when GitHub goes to call back to my deployment, I'll get an HTTP 400 error in the browser (not browser specific). when I look at the ATC logs, I see this:

```json
{"timestamp":"1533702640.815021753","source":"atc","message":"atc.dex.event","log_level":1,"data":{"fields":{},"message":"login successful: connector \"github\", name=\"Mike Lloyd\", email=\"mike@...\", groups=[\"...\"]","session":"5"}}
```

So I know it's properly receiving some information from GitHub. what's really interesting, when I start digger deeper, I'm seeing this error:

```json
{"timestamp":"1533702670.825408459","source":"atc","message":"atc.sky.callback.failed-to-fetch-dex-token","log_level":2,"data":{"error":"Post https://some.domain/sky/issuer/token: dial tcp 1.2.3.4:443: i/o timeout","session":"4.57"}}
```

What's really interesting, is if I run `curl -XPOST https://some.domain/sky/issuer/token` in a loop during that same exact time frame, it resolves fine and connects without a hitch:

```sh
22:31:09.821: POST /sky/issuer/token {"error":"invalid_client","error_description":"Invalid client credentials."}
...
22:31:10.040: POST /sky/issuer/token {"error":"invalid_client","error_description":"Invalid client credentials."}
...
22:31:10.384: POST /sky/issuer/token {"error":"invalid_client","error_description":"Invalid client credentials."}
```

To add on top of the mystery, both the ATC and curl are taking the exact same network path (ext network -> nginx -> int network), using the same DNS server, and hitting the same endpoint. curl doesn't have any connection errors or timeouts at all, it's very specifically the ATC which times out trying to connect to the `/sky/issuer/token` endpoint. any ideas?

The following can also be handy:

* Concourse version: 4.0.0
* Deployment type (BOSH/Docker/binary): bosh
* Infrastructure/IaaS: vSphere
* Browser (if applicable): Any
* Did this used to work? In 3.x

**Comment by vito** [comment]
Sounds like a duplicate of #2463 - we've got a fix in the works, ready for 4.1

**Comment by UniqueElphie** [comment]
Hey @vito I just setup v4.1, but when I try to login as local user or github user, the login fails due to the same error. Is this issue fixed and verified?
The relevant logging when I try to login as local user:
`{"timestamp":"1536016247.228306532","source":"atc","message":"atc.dex.event","log_level":1,"data":{"fields":{},"message":"login successful: connector \"local\", username=\"concourse\", email=\"concourse\", groups=[]","session":"5"}}
{"timestamp":"1536016249.150294065","source":"atc","message":"atc.sky.userinfo.failed-to-parse-authorization-header","log_level":1,"data":{"session":"4.104"}}
{"timestamp":"1536016254.152165890","source":"atc","message":"atc.sky.userinfo.failed-to-parse-authorization-header","log_level":1,"data":{"session":"4.105"}}
{"timestamp":"1536016255.191825628","source":"atc","message":"atc.build-tracker.track.start","log_level":0,"data":{"session":"19.48"}}
{"timestamp":"1536016255.195375443","source":"atc","message":"atc.build-tracker.track.done","log_level":0,"data":{"session":"19.48"}}
{"timestamp":"1536016257.772825003","source":"atc","message":"atc.sky.callback.failed-to-fetch-dex-token","log_level":2,"data":{"error":"Post https://externalurl/sky/issuer/token: net/http: TLS handshake timeout","session":"4.103"}}
{"timestamp":"1536016259.153844595","source":"atc","message":"atc.sky.userinfo.failed-to-parse-authorization-header","log_level":1,"data":{"session":"4.106"}}`
